### ISSUE TITLE
Issue running AppImage/Binary update on Fedora 38

### ISSUE THREAD

**Issue by sabinrimni** [dragoonDorise/EmuDeck#788]
When I run the `Update AppImage and Binaries` option from the UI, I get a list of things to update and upon clicking *OK*, a dialog flashed for a few seconds and then disappears. Nothing else happens after that.

In the log found in `/home/<user>/emudeck/Emudeck.AppImage.log`, the last entry is:
`[06/06/2023] bash ~/.config/EmuDeck/backend/tools/binupdate/binupdate.sh`
And nothing else after that.

I then ran that command manually from the terminal and got the following error:
```
/home/<user>/.config/EmuDeck/backend/tools/binupdate/binupdate.sh: line 283: 13353 Broken pipe             runBinDownloads "$binsToDL"
     13354 Segmentation fault      (core dumped) | zenity --progress --pulsate --title="Updating!" --width=600 --height=250 2> /dev/null

```

Just out of curiosity I fiddled a bit with the `binupdate.sh`, and if I change *line 262* (as of v2.1.4) from:
`runBinDownloads "$binsToDL" | zenity --progress --pulsate --title="Updating!" --width=600 --height=250 2>/dev/null`
to
`runBinDownloads "$binsToDL"`
It works, everything updated just fine (without the general progress bar of course). The bad news is that no matter how much I fiddled with the zenity command, I couldn't properly fix it. I suspect it has something to do with the latest version of zenity, although I couldn't find anything relevant in their [changelog](https://gitlab.gnome.org/GNOME/zenity/-/blame/master/NEWS).

EmuDeck version: 2.1.4
OS: Fedora 38
Zenity version: 3.92.0

**Comment by Godsbane** [comment]
This def sounds like an upstream zenity bug.

If you can downgrade it and retest. Report the bug with progress to zenity or fedora depending on where the issue actually is.

As it works fine on the deck, I don't think this is a usage issue on our side. I can do some arch testing later.

**Comment by sabinrimni** [comment]
Yeah, I just checked and the Deck is running zenity 3.32.0 (as of SteamOS 3.4.8) which is 4 years old at this point. I opened an [issue](https://gitlab.gnome.org/GNOME/zenity/-/issues/61) on the zenity project and will leave this one open so I can reference it.

**Comment by Godsbane** [comment]
Thanks for doing that. When I have some time I will test on my Nobara and endeavor boxes as well.

**Comment by Godsbane** [comment]
@sabinrimni - did you look at the other issue opened for similar? https://gitlab.gnome.org/GNOME/zenity/-/issues/60

that person was using zsh and had trouble - can you validate which shell you used? maybe explicitly use bash with the script and see if the issue persists?

**Comment by cfgnunes** [comment]
I have the same problem using Fedora 38 running zenity 3.92.0. I think it is a zenity bug.

This bug crashes my "Nautilus Scripts" in: https://github.com/cfgnunes/nautilus-scripts

There, I use:
```bash
cat "$TEMP_FIFO" | (
    zenity \
        --title="$(_get_script_name)" \
        --width=400 \
        --progress \
        --pulsate \
        --auto-close \
        --text="$message" ||
        (echo >"$TEMP_FIFO" && _exit_script)
) &
```

**Comment by sabinrimni** [comment]
Fedora has now bumped zenity to 3.99.1 and with that the issue has been fixed and the script runs as intended from the UI. The zenity maintainer also suggests in their [issue](https://gitlab.gnome.org/GNOME/zenity/-/issues/61) that it was fixed somewhere in 3.99.0 (mentioning this for future reference). With that, I'm closing this issue.
